frontend/src/modules/Settings/Components/Settings.tsx

```diff
@@ -1,24 +1,87 @@
 import { DropdownSelect } from '@core/index'
-import { Box, IconButton, SelectChangeEvent } from '@mui/material'
-import Menu from '@mui/material'
+import { Box, IconButton, SelectChangeEvent, Switch } from '@mui/material'
 import MenuItem from '@mui/material/MenuItem'
 import { useState } from 'react'
 import { Link } from 'react-router-dom'
 import { DASHBOARD_PATH } from '@core/index'
 import { ReactComponent as LogoImg } from '@assets/img/lscicon.svg'
 import { AccountMenu } from 'Dashboard/Components/AccountMenu'
+import { API_ROOT, COURSE_API } from '@core/Constants'
+import { AdministratorsTable } from './AdministratorsTable'
+import { Admin } from './types'
+import axios from 'axios'
 
 export const Settings = () => {
   const [currRoster, setCurrRoster] = useState<string>('SP23')
-  const semesters = ['SU22', 'FA22', 'WI23', 'SP23', 'SU23']
-
   const changeCurrRoster = (event: SelectChangeEvent) => {
     // function to only open the survey of the semester
     setCurrRoster(event.target.value)
   }
+
+  const [semesters, setSemesters] = useState<string[]>([])
+  function getAllSemesters() {
+    axios
+      .get(`${API_ROOT}${COURSE_API}/semester/all`)
+      .then((res) => setSemesters(res.data))
+  }
+  getAllSemesters()
+
+  const [surveyState, setSurveyState] = useState<boolean>()
+
+  const changeSurveyAvailability = () => {
+    axios.post(`${API_ROOT}${COURSE_API}/semester/survey`, {
+      surveyOpen: !surveyState,
+    })
+    setSurveyState(!surveyState)
+  }
+
+  const getCurrSurveyState = async () => {
+    await axios.get(`${API_ROOT}${COURSE_API}/semester/survey`).then((req) => {
+      setSurveyState(req.data)
+    })
+  }
+
+  const [administrators, setAdministrators] = useState<Admin[]>([])
+
+  const getAdministrators = async () => {
+    await axios.get(`${API_ROOT}/admin`).then((req) => {
+      setAdministrators(req.data)
+    })
+  }
+
+  const removeAdmin = (admin: Admin) => {
+    axios
+      .delete(`${API_ROOT}/admin`, { data: admin })
+      .then(() => {
+        administrators.forEach((email, index) => {
+          if ((email.email = admin.email)) {
+            administrators.splice(index, 1)
+            getAdministrators()
+          }
+        })
+      })
+      .catch((err) => console.log(err))
+  }
+
+  // TODO: allow to edit admin information
+  const editAdmin = () => {}
+
+  // grabs the state of the backend once when you refresh code
+  const [start, setStart] = useState<boolean>(true)
+  if (start) {
```

Comment:
Instead of doing this, it would be better to also put this in an useEffect hook.